### ISSUE TITLE
OS X library missing

### ISSUE THREAD

**Issue by Byteron** [Beliaar/godot-yaml-asset#3]
Hello there!
It seems there is a problem on OS X platforms. We get this error when trying to run our project:
```
0:00:00:0345 - Can't open dynamic library: /Users/Emilien/Documents/Dev/haldric/addons/godot-yaml/osx/libgodotyaml.dylib. Error: dlopen(/Applications/Godot.app/Contents/MacOS/../Frameworks/libgodotyaml.dylib, 2): image not found
----------
Type :Erreur
Description : Can't open dynamic library: /Users/Emilien/Documents/Dev/haldric/addons/godot-yaml/osx/libgodotyaml.dylib. Error: dlopen(/Applications/Godot.app/Contents/MacOS/../Frameworks/libgodotyaml.dylib, 2): image not found
Temps : 0:00:00:0345
C Erreur : Method/Function Failed, returning: ERR_CANT_OPEN
C Source : platform/osx/os_osx.mm:1346
C Fonction : open_dynamic_library
0:00:00:0345 - No valid library handle, can't get symbol from GDNative object
----------
Type :Erreur
Description : 
Temps : 0:00:00:0345
C Erreur : No valid library handle, can't get symbol from GDNative object
C Source : modules/gdnative/gdnative.cpp:317
C Fonction : get_symbol
0:00:00:0345 - No nativescript_init in "res://addons/godot-yaml/osx/libgodotyaml.dylib" found
----------
Type :Erreur
Description : 
Temps : 0:00:00:0345
C Erreur : No nativescript_init in "res://addons/godot-yaml/osx/libgodotyaml.dylib" found
C Source : modules/gdnative/nativescript/nativescript.cpp:1054
C Fonction : init_library
```

**Comment by Byteron** [comment]
I see, apparently there is no library for OS X in the first place.
Would it be possible to add it?

**Comment by Beliar83** [comment]
Heya, I do not have access to OS X myself, and the build service I use, appveyor, does also not have support for that, I think.

I would be happy to add the libraries if I had them.

**Comment by Byteron** [comment]
in your godot-yaml repo there seems to be this part in the Sconstruct file. 
```
if platform == "osx":
    env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64', '-std=c++14'])
    env.Append(LINKFLAGS = ['-arch', 'x86_64'])
    env.Append(LIBS=["libyaml-cpp"])
final_lib_path = final_lib_path + 'osx/'
```
I thought it indicated that building for osx is possible. But then again, I have no idea how scons works.

**Comment by Beliar83** [comment]
Yes, that is true, but as far as I know this will not work unless one has an OSX (virtual) machine that can build that. And as said, appveyor has, as far as I know, no images for OSX. I will try to check if I can find a build service for that in the next days. I would also take suggestions, if you know one.

**Comment by Byteron** [comment]
So if I had an os x system, I could build it?

**Comment by Beliar83** [comment]
Yes, if you have the tools (compiler, scons) setup. You can look up the details here: https://docs.godotengine.org/en/3.0/tutorials/plugins/gdnative/gdnative-cpp-example.html#compiling-our-plugin

Also check the readme in the godot-yaml repo, even though it is probably not explaining much.

**Comment by Byteron** [comment]
http://docs.godotengine.org/en/3.0/development/compiling/compiling_for_osx.html#cross-compiling

**Comment by Beliar83** [comment]
Interesting, I will take a look at that, I may be able to do that on the linux image in appveyor. Thanks.

**Comment by Beliar83** [comment]
I have uploaded the osx library. I will update the info of the godot asset library shortly, which then needs to be approved by someone from godot.

Please reopen if it does not work with the added library.

**Comment by critcap** [comment]
The osx library doesnt seemt to work with godot 3.2.1 atleast. Getting the exact same error as the original poster.

**Comment by Beliar83** [comment]
@critcap Can you confirm whether the file "addons\godot-yaml\osx\libgodotyaml.dylib" exists?

There is also a possibility that the current files do not work with godot 3.2.1, as they were made for godot 3.1 and I have not worked on this since then.

**Comment by critcap** [comment]
Yes, its in the osx folder. I also tried to move it to other locations and changing the path but nothing helped. 
It might be an incompatibility with godot 3.2.1, are there any plans to update?

**Comment by Beliar83** [comment]
@critcap I will look into it. Hopefully not much changed in terms of native.

**Comment by Beliar83** [comment]
@critcap I have uploaded a version for 3.2 and submitted a new asset. You can either wait for someone to approve it, or download the latest release from here.

**Comment by critcap** [comment]
still doesnt work for me on osx, same error as before. Are there any possbile prerequisites im missing?
My error message:
<img width="1051" alt="Screenshot 2020-03-19 at 16 40 29" src="https://user-images.githubusercontent.com/39470937/77085265-5f526100-6a00-11ea-92b4-d18cfcc7d9df.png">

**Comment by Beliar83** [comment]
There should be no prerequisites. I am using cross compiling on linux, though, maybe it is just not working. I don't have osx myself, though, so I can't really test it myself.

I remember reading that travis has added building on osx recently, though, I will take a look at that.

**Comment by Beliar83** [comment]
Ah, actually, that is why I am already using travis for osx builds, so nothing to change there.

**Comment by Beliar83** [comment]
@critcap It is possible, though, that you may need to install the "yaml-cpp" package, not sure if it is linked dynamcially or statically.

**Comment by C45tr0** [comment]
I am having the same issue, except on windows. Same error except instead of the osx path its "{...}/addons/godot-yaml/win64/libgodotyaml.dll".

I tried building the yaml-cpp package. Not sure if I installed it or not(might be the issue, still digging into it), but still getting the same errors.

I tried the asset store, then tried the files in this repo in case they were still outdated on the asset store.

**Comment by Beliar83** [comment]
I can't reproduce that here. I am going to setup up VM to see if I can reproduce them there.

**Comment by Beliar83** [comment]
@C45tr0 I cannot reproduce that even on a fresh windows (10) install.
Is it possible, that windows flagged the DLL files as insecure - which may happen for downloaded files - and you need to set them as secure? Otherwise I am not sure what the problem can be.

**Comment by C45tr0** [comment]
Thanks for looking.

I looked at permissions and didn't see anything off. Will keep trying some things tonight.

**Comment by shayneoneill** [comment]
I just noticed when installing it, it tries to skip installing the dll/dylib/etc. Make sure you actually tick them if installing from app store

edit: Nope that dont fix it
